diff --git a/backend/src/lib/auth.ts b/backend/src/lib/auth.ts
--- a/backend/src/lib/auth.ts
+++ b/backend/src/lib/auth.ts
@@ -22,7 +22,10 @@ const scope = "openid email profile org.cilogon.userinfo";
 const allowedIdps = parseCsv(env.ALLOWED_IDPS);
 
 const getIdentity = (claims: client.IDToken) => {
-  return claims[env.LOGIN_CLAIM] as string;
+  // Rancher is not configured to use a claim available from CILogon as a principalId.
+  // Rather, it uses the Purdue-specific UID:
+  const email = claims.email as string;
+  return email.replace("@purdue.edu", "");
 };
 
 const router = express.Router();
