diff --git a/node_modules/@fishballpkg/acme/_dist/src/AcmeChallenge.d.ts b/node_modules/@fishballpkg/acme/_dist/src/AcmeChallenge.d.ts
index 43b2f48..6b2587b 100755
--- a/node_modules/@fishballpkg/acme/_dist/src/AcmeChallenge.d.ts
+++ b/node_modules/@fishballpkg/acme/_dist/src/AcmeChallenge.d.ts
@@ -63,6 +63,7 @@ import type { AcmeAuthorization } from "./AcmeAuthorization.js";
   /**
    * Fetches a snapshot of the challenge object from the Certificate Authority (CA).
    */ fetch(): Promise<AcmeChallengeObjectSnapshot>;
+  keyAuthorization(): Promise<string>;
   /**
    * Produces the key authorization for this challenge by digesting the challenge token.
    */ digestToken(): Promise<string>;
diff --git a/node_modules/@fishballpkg/acme/_dist/src/AcmeClient.d.ts b/node_modules/@fishballpkg/acme/_dist/src/AcmeClient.d.ts
index 81dff96..463314e 100755
--- a/node_modules/@fishballpkg/acme/_dist/src/AcmeClient.d.ts
+++ b/node_modules/@fishballpkg/acme/_dist/src/AcmeClient.d.ts
@@ -1,3 +1,4 @@
+import type { webcrypto } from "node:crypto"; // AnvilOps patch: fix type error
 import { AcmeAccount } from "./AcmeAccount.js";
 /**
  * The directory object.
@@ -40,7 +41,7 @@ import { AcmeAccount } from "./AcmeAccount.js";
   /**
    * Fetch a url where the data are signed with the provided private key using JSON Web Signature (JWS).
    */ jwsFetch(url: string, {}: {
-    privateKey: CryptoKey;
+    privateKey: webcrypto.CryptoKey; // AnvilOps patch: fix type error
     protected: Record<PropertyKey, unknown>;
     payload?: Record<PropertyKey, unknown>;
     retryAttemptCount?: number;
diff --git a/node_modules/@fishballpkg/acme/src/AcmeChallenge.js b/node_modules/@fishballpkg/acme/src/AcmeChallenge.js
index 054b806..f8cc822 100755
--- a/node_modules/@fishballpkg/acme/src/AcmeChallenge.js
+++ b/node_modules/@fishballpkg/acme/src/AcmeChallenge.js
@@ -38,11 +38,14 @@ import { encodeBase64Url } from "./utils/encoding.js";
     const response = await this.#account.jwsFetch(this.url);
     return await response.json();
   }
+  async keyAuthorization() {
+    const publicKeyJwk = await crypto.subtle.exportKey("jwk", this.#account.keyPair.publicKey);
+    return `${this.token}.${await getJWKThumbprint(publicKeyJwk)}`;
+  }
   /**
    * Produces the key authorization for this challenge by digesting the challenge token.
    */ async digestToken() {
-    const publicKeyJwk = await crypto.subtle.exportKey("jwk", this.#account.keyPair.publicKey);
-    return encodeBase64Url(await crypto.subtle.digest("SHA-256", new TextEncoder().encode(`${this.token}.${await getJWKThumbprint(publicKeyJwk)}`)));
+    return encodeBase64Url(await crypto.subtle.digest("SHA-256", new TextEncoder().encode(await this.keyAuthorization())));
   }
   /**
    * Submit the challenge.
diff --git a/node_modules/@fishballpkg/acme/src/AcmeChallenge.ts b/node_modules/@fishballpkg/acme/src/AcmeChallenge.ts
index 2fe7b5a..cc12af2 100755
--- a/node_modules/@fishballpkg/acme/src/AcmeChallenge.ts
+++ b/node_modules/@fishballpkg/acme/src/AcmeChallenge.ts
@@ -106,6 +106,11 @@ export class AcmeChallenge {
     return await response.json();
   }
 
+  async keyAuthorization(): Promise<string> {
+    const publicKeyJwk = await crypto.subtle.exportKey("jwk", this.#account.keyPair.publicKey);
+    return `${this.token}.${await getJWKThumbprint(publicKeyJwk)}`;
+  }
+
   /**
    * Produces the key authorization for this challenge by digesting the challenge token.
    */
@@ -153,7 +158,7 @@ export class AcmeChallenge {
   }
 }
 
-async function getJWKThumbprint(jwk: JsonWebKey): Promise<string> {
+export async function getJWKThumbprint(jwk: JsonWebKey): Promise<string> {
   // Step 1: Create the canonical JSON string from required JWK fields,
   const canonicalJWK = JSON.stringify({
     crv: jwk.crv,
diff --git a/node_modules/@fishballpkg/acme/src/utils/encoding.js b/node_modules/@fishballpkg/acme/src/utils/encoding.js
index 85bf5c4..fd8412f 100755
--- a/node_modules/@fishballpkg/acme/src/utils/encoding.js
+++ b/node_modules/@fishballpkg/acme/src/utils/encoding.js
@@ -1,16 +1,20 @@
+// AnvilOps patch: we're using a polyfill here. This method was added to Node in version 25, which will go out of support soon, and 26 isn't out yet.
+// This patch can be removed after upgrading to Node.js 26 or later.
+import toBase64 from "es-arraybuffer-base64/Uint8Array.prototype.toBase64";
+
 const toUint8Array = (input)=>{
   if (input instanceof Uint8Array) return input;
   if (input instanceof ArrayBuffer) return new Uint8Array(input);
   return new TextEncoder().encode(input);
 };
 export const encodeBase64Url = (input)=>{
-  return toUint8Array(input).toBase64({
+  return toBase64(toUint8Array(input), {
     alphabet: "base64url",
     omitPadding: true
   });
 };
 export const encodeBase64 = (input)=>{
-  return toUint8Array(input).toBase64({
+  return toBase64(toUint8Array(input), {
     alphabet: "base64"
   });
 };
diff --git a/node_modules/@fishballpkg/acme/src/utils/pem.js b/node_modules/@fishballpkg/acme/src/utils/pem.js
index bdf0dd0..11adf20 100755
--- a/node_modules/@fishballpkg/acme/src/utils/pem.js
+++ b/node_modules/@fishballpkg/acme/src/utils/pem.js
@@ -1,2 +1,2 @@
-export const extractFirstPemObject = (pem)=>Uint8Array.fromBase64(pem.replace(/-----BEGIN [^-]+-----/, "").replace(/-----END [^-]+-----.*$/s, "").replaceAll(/\s/g, ""));
+export const extractFirstPemObject = (pem)=>Uint8Array.from(Buffer.from(pem.replace(/-----BEGIN [^-]+-----/, "").replace(/-----END [^-]+-----.*$/s, "").replaceAll(/\s/g, ""), "base64"));
 //# sourceMappingURL=pem.js.map
\ No newline at end of file
