diff --git a/node_modules/openapi-backend/backend.js b/node_modules/openapi-backend/backend.js
index 330013d..8604e93 100644
--- a/node_modules/openapi-backend/backend.js
+++ b/node_modules/openapi-backend/backend.js
@@ -365,6 +365,13 @@ class OpenAPIBackend {
                 }
                 return notImplementedHandler(context, ...handlerArgs);
             }
+
+            if (this.validator.coerceTypes && validate) {
+                context.request.query = context.validation.coerced.query;
+                context.request.params = context.validation.coerced.params;
+                context.request.headers = context.validation.coerced.headers;
+            }
+
             // handle route
             return operationHandler(context, ...handlerArgs);
         }).bind(this)();
diff --git a/node_modules/openapi-backend/validation.js b/node_modules/openapi-backend/validation.js
index c76f207..28b78cc 100644
--- a/node_modules/openapi-backend/validation.js
+++ b/node_modules/openapi-backend/validation.js
@@ -174,7 +174,7 @@ class OpenAPIValidator {
      * @memberof OpenAPIRequestValidator
      */
     validateRequest(req, operation) {
-        const result = { valid: true, coerced: { ...req } };
+        const result = { valid: true, coerced: { ...req, path: req.path } };
         result.errors = [];
         if (!operation) {
             operation = this.router.matchOperation(req);
@@ -244,6 +244,7 @@ class OpenAPIValidator {
             else if (this.coerceTypes) {
                 result.coerced.query = parameters.query;
                 result.coerced.params = parameters.path;
+                result.coerced.headers = parameters.header;
             }
         }
         if (_.isEmpty(result.errors)) {
@@ -452,9 +453,9 @@ class OpenAPIValidator {
         // schema for operation requestBody
         if (operation.requestBody) {
             const requestBody = operation.requestBody;
-            const jsonbody = requestBody.content['application/json'] ||
-                requestBody.content['multipart/form-data'] ||
-                requestBody.content['application/x-www-form-urlencoded'];
+            // AnvilOps patch: revert this commit: https://github.com/openapistack/openapi-backend/commit/aa2549601bf4936013185fbe66bb388a81d65f64
+            // due to issues with multipart/form-data requests not having a requestBody property
+            const jsonbody = requestBody.content['application/json'];
             if (jsonbody && jsonbody.schema) {
                 const requestBodySchema = {
                     title: 'Request',
